platform-service-impl/src/main/java/esa/mo/platform/impl/provider/opssat/PowerControlOPSSATAdapter.java

```diff
@@ -123,17 +145,66 @@ public void enableDevices(final DeviceList inputList) throws IOException
       }
       if (payloadId != null) {
         switchDevice(payloadId, device.getEnabled());
+        // When device is set OFF, mark the status right away
+        if(!device.getEnabled())
+        {
+          Device d = findByType(device.getDeviceType());
+          d.setEnabled(false);
+          if(d.getDeviceType() == DeviceType.ADCS){
+            adcsChannelStartTime = null;
+          }
+        }
       } else {
         throw new IOException("Cannot find the device by oId " + device);
       }
     }
   }
 
-  private void switchDevice(final PayloadDevice device, final Boolean enabled) throws IOException
+
+  @Override
+  public boolean isDeviceEnabled(DeviceType deviceType) {
+    boolean isEnabled = findByType(deviceType) == null ? false : findByType(deviceType).getEnabled();
+
+    // In the ADCS case, check that the power channel has been running at least a minute to get an accurate reading
+    if( deviceType.equals(DeviceType.ADCS) && (System.currentTimeMillis() - adcsChannelStartTime < 60*1000))
+    {
+       isEnabled = false;
+    }
+    return isEnabled;
+  }
+
+  @Override
+  public void startStatusTracking(ConnectionConsumer connection) {
+    CompleteDataReceivedListener listener = new CompleteDataReceivedListener() {
+      @Override
+      public void onDataReceived(ParameterInstance parameterInstance) {
+        if (parameterInstance == null) {
+          LOGGER.log(Level.WARNING, "Received a null parameter instance");
+          return;
+        }
+
+        Attribute rawValue = parameterInstance.getParameterValue().getRawValue();
+        String str = HelperAttributes.attribute2string(rawValue);
```

Comment:
`PDU1952` is of UShort type - its value can be retrieved directly, e.g. `long val = ((UShort)rawValue).getValue();`

Reply:
Done.